#! /usr/bin/env python2
# /****************************************************************************
#  *   Copyright (c) 2017 Michael Shomin. All rights reserved.
#  *
#  * Redistribution and use in source and binary forms, with or without
#  * modification, are permitted provided that the following conditions
#  * are met:
#  *
#  * 1. Redistributions of source code must retain the above copyright
#  *    notice, this list of conditions and the following disclaimer.
#  * 2. Redistributions in binary form must reproduce the above copyright
#  *    notice, this list of conditions and the following disclaimer in
#  *    the documentation and/or other materials provided with the
#  *    distribution.
#  * 3. Neither the name ATLFlight nor the names of its contributors may be
#  *    used to endorse or promote products derived from this software
#  *    without specific prior written permission.
#  *
#  * NO EXPRESS OR IMPLIED LICENSES TO ANY PARTY'S PATENT RIGHTS ARE GRANTED BY THIS LICENSE.
#  * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
#  * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
#  * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
#  * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
#  * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
#  * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
#  * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
#  * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
#  * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
#  * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
#  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
#  * POSSIBILITY OF SUCH DAMAGE.
#  *
#  * In addition Supplemental Terms apply.  See the SUPPLEMENTAL file. 
#  ****************************************************************************/


from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

'''
#TODO: implement params that require restarting the camera driver (level 3
gen.add("width", int_t, 3, "Image width in pixels", 640, 0)
gen.add("height", int_t, 3, "Image height in pixels", 480, 0)
gen.add("frame_rate", double_t, 3, "Frames per second", 30.0, 1.0, 120.0)

#TODO: implement this as a param.
gen.add("camera_info_url", str_t, 0, "Path to camera calibration file.", "")
'''

gen.add("frame_id", str_t, 0,
        "ROS tf frame of reference, resolved with tf_prefix unless absolute.",
        "camera")

gen.add("frame_id_right", str_t, 0,
        "ROS tf frame of reference, resolved with tf_prefix unless absolute.",
        "right")

gen.add("gain", double_t, 0,"Gain, device dependent.", 0.15, 0, 1.0)
gen.add("exposure", double_t, 0,"exposure, device dependent.", 0.5, 0, 1.0) 
gen.add("skip_n_frames", int_t, 0, "Skip N frames to throttle publishing.", 0, 0, 30)


exit(gen.generate('snap_cam_ros', "snap_cam_ros", "SnapCam"))
